dtls.c

```diff
@@ -190,6 +190,34 @@ memarray_t dtlscontext_storage;
     P += dtls_ ## T ## _to_int(P) + sizeof(T);				\
   }
 
+/* The length check here should work because dtls_*_to_int() works on
+ * unsigned char. Otherwise, broken messages could cause severe trouble.
+ * Note that this macro returns an error when the message is too short.
+ *
+ * \param R  result, variable to assign the length of the field.
```

Comment:
> I think that R gets assigned the contents of the field P points to, not its length (cf. line 212). 

That may depend more on the used terms and their definition.
In my definition of `field` this not a term for internal structs, this is used for a "logical value".
In the case of a "variable length field", that `field` is split into it's `length` and `value`, but I don't use the term `field`for this parts. What this macro does is reading that `length` of the `field` into R and it also forwards the point P to the `value` of that `field`.  I applies also all length checks in order to protect from "out of bounds" errors.

> At least, this comment is misleading, or the macro should be called something like GET_VAR_FIELD_LENGTH as it is used to get the length value of a variable-length TLS header field.

Though it does more, that name isn't really a that better match.
For me, it prepares all, including checks and failures returns, in order to the straight forward process the value of the `field`. 

I also consider to change the code even more and put that in a function. But then all calls would need to check the return and conditionally return on failures. So I decided to go for this macro, though it seems for me to be easier to read the resulting code.

Reply:
> In my definition of `field` this is not a term for internal structs, this is used for a "logical value". In the case of a "variable length field", that `field` is split into its `length` and `value`, but I don't use the term `field`for this parts. What this macro does is reading that `length` of the `field` into R and it also forwards the point P to the `value` of that `field`. I applies also all length checks in order to protect from "out of bounds" errors.

Yes, but it would be good to put this explanation in the comment for this macro as it helps understanding what the macro does.

Reply:
OK.

Reply:
The comment has been edited.